build: fix vendor and vendor datadriven

tbg opened the pull request:
The github.com/cockroachdb/datadriven package is identical to the
code deleted in this commit.

Also replaced a SHA for honnef.co/go/tools. The original SHA is
no longer available (the new one is equivalent, taken from the
master branch so it's hopefully going to last).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/38109)
<!-- Reviewable:end -->

RaduBerinde commented on Gopkg.toml:
```diff
@@ -101,7 +101,7 @@ ignored = [
   name = "honnef.co/go/tools"
   # This revision is not yet on master (as of 05-12-2019); when it is, we can
```
Update comment. We should just set it to master instead of a fixed sha.

tbg replied:
I used `branch = "master"` instead but then various things break because there have been API changes. I updated the comment but I'll leave it pinned at the revision since I just wanted to fix vendoring, not bump this. Let me know if I should've done something different and I'll follow up.

tbg committed:
build: fix vendor and vendor datadriven

The github.com/cockroachdb/datadriven package is identical to the
code deleted in this commit.

Also replaced a SHA for honnef.co/go/tools. The original SHA is
no longer available (the new one is equivalent, taken from the
master branch so it's hopefully going to last).

Release note: None

tbg commented:
bors r=RaduBerinde
